org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java

```diff
@@ -248,6 +256,8 @@ private void startConnection() throws IOException {
 	 */
 	@Override
 	public void stop(BundleContext bundleContext) throws Exception {
+		CorePlugin.listenerRegistry().dispatch(new WorkbenchShutdownEvent());
```

Comment:
please keep the JavaLanguageServerPlugin decoupled from buildship

Reply:
Feels wrong to have to send a workbenchshutdownevent here, specifically to the buildship plugin. That should be taken care of automatically by the platform.

Reply:
It is just a marker event. It is only used to save a model. See https://github.com/eclipse/buildship/blob/master/org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/internal/DefaultModelPersistence.java#L91